src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -2131,8 +2308,49 @@ public void testItemHasLastCheckInProperties()
     assertThat(expected.getStaffMemberId(), is(actual.getStaffMemberId()));
   }
 
-   private Response getById(UUID id)
-    throws MalformedURLException, InterruptedException,
+  @Test
+  public void cannotCreateItemWithWrongStatus() throws Exception {
+    JsonObject itemToCreate = new JsonObject()
+      .put("id", UUID.randomUUID().toString())
+      .put("status", new JsonObject().put("name", "Wrong status name"));
+
+    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+    client.post(itemsStorageUrl(""), itemToCreate, StorageTestSuite.TENANT_ID,
+      ResponseHandler.text(createCompleted));
+
+    Response postResponse = createCompleted.get(5, TimeUnit.SECONDS);
+    assertThat(postResponse.getStatusCode(), is(400));
+    assertThat(postResponse.getBody(),
+      matchesPattern("(?s)Json content error Cannot construct instance of `.+`, problem: Wrong status name.+")
+    );
+  }
+
+  @Test
+  public void canRemoveItemStatus() throws Exception {
```

Comment:
Why are we allowing an item status to be removed?

Reply:
Well, the status is not a required property, so I assumed it is possible case.

Reply:
I'm wondering if it might be worth making it required as part of this change. 

It shouldn't be removable, it is only marked not required because it is defaulted during initial creation.

Reply:
Do we need to declare it in schema or do it in code?

We won't be allowed to do the defaulting through JsonSchema if the entire `status` field  is not present, it will be possible for `name` field only.

For example:
1.
```
{
  "id": "id",
  "hrid": "hrid"
  ... <no-status>
}
```
Won't set a default value and will throw validation error.

2. 
```
{
  "id": "id",
  "hrid": "hrid",
  "status": {}
}
```
This case will work, we will be able to set a default value for name.

Reply:
The default is done in code. (I don't find defaults in schema particularly useful).

I think the defaulting should be removed as part of this change and instead the status be required to be set by the client.

We have had issues with defaulting values, and don't really want to do it unless absolutely necessary. This was likely done trying to be helpful or avoid a breaking change, I don't think it adds much value at this stage.

Reply:
Ok, I've removed the defaulting logic and declared the status as required field.